e2store/src/utils.rs

```diff
@@ -47,10 +48,11 @@ pub async fn download_era_links(
 pub async fn get_era_files(http_client: &Client) -> anyhow::Result<HashMap<u64, String>> {
     let era_files = download_era_links(http_client, ERA_DIR_URL).await?;
     ensure!(!era_files.is_empty(), "No era files found at {ERA_DIR_URL}");
-    ensure!(
-        (0..era_files.len()).all(|epoch| era_files.contains_key(&(epoch as u64))),
-        "Epoch indices are not starting from zero or not consecutive",
-    );
+    for epoch_num in 0..era_files.len() {
+        if !era_files.contains_key(&(epoch_num as u64)) {
+            warn!("Epoch indices are not starting from zero or not consecutive: missing epoch {epoch_num}");
```

Comment:
This shouldn't be a warning. This code is used elsewhere. It isn't only for the CI. It ensures whoever calls this function gets a complete sequential era_file list, the warning would be unexpected behavior from a user's perspective.